### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by genofire**
for some questions

All around: nice :+1:

### CODE REVIEW

File: src/metrics.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    COUNTER_DHCPACK,
+    COUNTER_DHCPDECLINE,
+    COUNTER_DHCPDISCOVER,
+    COUNTER_DHCPINFORM,
+    COUNTER_DHCPNAK,
+    COUNTER_DHCPOFFER,
+    COUNTER_DHCPRELEASE,
+    COUNTER_DHCPREQUEST,
+    COUNTER_NOANSWER,
+    COUNTER_MAX,
```
Comment (genofire) [inline src/metrics.h]: should be a `__COUNTER_MAX` better?
It would more obviously that this is no real counter.

File: src/ubus.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+	poll_listen(ubus->sock.fd, POLLIN);
+	poll_listen(ubus->sock.fd, POLLERR);
+	poll_listen(ubus->sock.fd, POLLHUP);
+}
+
+void check_ubus_listeners()
+{
+	if (!ubus) {
+		ubus = ubus_connect(NULL);
```
Comment (genofire) [inline src/ubus.c:55]: should not be `ubus_connect` in `set_ubus_listeners` ?

File: src/metrics.h (hunk 1 above)
Comment (corny) [reply src/metrics.h]: :+1:

**Comment by stappersg** [comment]
See also http://lists.thekelleys.org.uk/pipermail/dnsmasq-discuss/2018q2/012130.html
That is a posting on the dnsmasq mailinglist with subject 'Re: [Dnsmasq-discuss] ubus and metrics'.

Snippet
<tt>
Some files have

METRICS_INCREMENT(<metric>);

and some have

#ifdef HAVE_METRICS
     metrics[<metric>]++;
#endif

Any good reason for that?
</tt>

**Comment by stappersg** [comment]
And yes, you are welcome on the  dnsmasq  mailinglist.
http://lists.thekelleys.org.uk/mailman/listinfo/dnsmasq-discuss

**Comment by corny** [comment]
@stappersg Thanks for your comments. In the meanwhile I updated my pull request.

**Comment by genofire** [comment]
Its upstream now ....

**Pull request closed without merging**